2/contrib/jenkins/install-plugins.sh

```diff
@@ -1,436 +1,8 @@
-            (cd $TEMP_PLUGIN_DIR;jar xf "$JENKINS_WAR" "$i";jar xvf "$TEMP_PLUGIN_DIR/$i" META-INF/MANIFEST.MF >/dev/null 2>&1)
-            VER=$(egrep -i Plugin-Version "$TEMP_PLUGIN_DIR/META-INF/MANIFEST.MF"|cut -d: -f2|sed 's/ //')
-            echo "$PLUGIN:$VER"
-        done
-        rm -fr $TEMP_PLUGIN_DIR
-    else
-        echo "ERROR file not found: $JENKINS_WAR"
-        exit 1
-    fi
-}
-
-function versionFromPlugin() {
-    local plugin=$1
-    if [[ $plugin =~ .*:.* ]]; then
-        echo "${plugin##*:}"
-    else
-        echo "latest"
-    fi
-
-}
-
-function installedPlugins() {
-    for f in "$REF_DIR"/*.jpi; do
-        echo "$(basename "$f" | sed -e 's/\.jpi//'):$(get_plugin_version "$f")"
-    done
-}
-
-function jenkinsMajorMinorVersion() {
-    if [[ -f "$JENKINS_WAR" ]]; then
-        local version major minor
-        version="$(/etc/alternatives/java -jar $JENKINS_WAR --version)"
-        major="$(echo "$version" | cut -d '.' -f 1)"
-        minor="$(echo "$version" | cut -d '.' -f 2)"
-        echo "$major.$minor"
-    else
-        echo "ERROR file not found: $JENKINS_WAR"
-        return 1
-    fi
-}
+source /usr/local/bin/install-plugins-internals.sh
 
 main() {
-    local plugin version
-
-    mkdir -p "$REF_DIR" || exit 1
-
-    for file in $@; do
-        # clean up any dos file injected carriage returns
-        sed -i 's/\r$//' $file
-    done
-
-    # Create lockfile manually before first run to make sure any explicit version set is used.
-    echo "Creating initial locks..."
-    for plugin in `cat $@ | grep -v ^#`; do
-        if [ -z $plugin ]; then
-            continue
-        fi
-        echo "Locking $plugin"
-        mkdir "$(getLockFile "${plugin%%:*}")"
-    done
-
-    echo -e "\nAnalyzing war..."
-    bundledPlugins="$(bundledPlugins)"
-    
-    # Check if there's a version-specific update center, which is the case for LTS versions
-    jenkinsVersion="$(jenkinsMajorMinorVersion)"
-    if curl -fsL -o /dev/null "$JENKINS_UC/$jenkinsVersion"; then
-        JENKINS_UC_LATEST="$JENKINS_UC/$jenkinsVersion"
-        echo "Using version-specific update center: $JENKINS_UC_LATEST..."
-    else
-        JENKINS_UC_LATEST=
-    fi
-
-    echo -e "\nDownloading plugins..."
-    for plugin in `cat $@ | grep -v ^#`; do
-        if [ -z $plugin ]; then
-            continue
-        fi
-        version=""
-        if [[ $plugin =~ .*:.* ]]; then
-            version=$(versionFromPlugin "${plugin}")
-            plugin="${plugin%%:*}"
-        fi
-
-        download "$plugin" "$version" "true"
-    done
-    wait
-
-    echo
-    echo "WAR bundled plugins:"
-    echo "${bundledPlugins}"
-    echo
-    echo "Installed plugins:"
-    installedPlugins
-
-    if [[ -f $FAILED ]]; then
-        echo -e "\nSome plugins failed to download!\n$(<"$FAILED")" >&2
-        exit 1
-    fi
-
-    echo -e "\nCleaning up locks"
-    rm -rf "$REF_DIR"/*.lock
+    mainInstallPlugins "$@"
 }
```

Comment:
doesn't seem like you need to define main() at all.  Just call mainInstallPlugins below.

Reply:
Yeah I wasn't sure re: bash nuances if that would somehow mess up existing s2i extensions that leverate install-plugins.sh 

Assuming you don't think of any in the next few minutes, I'll make this change